src/test/java/DecodeTest.java

```diff
@@ -0,0 +1,112 @@
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import decoder.Decoder;
+import encoder.Encoder;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Validate that decoding the protobuf file to json works properly. */
+@RunWith(JUnit4.class)
+public final class DecodeTest {
+  private static InputStream encoderInput;
+  private static OutputStream encoderOutput;
+  private static String decoderOutputFile;
+  private static InputStream decoderInput;
+  private static OutputStream decoderOutput;
+  private static String[] testInputArray;
+  private static String testInput;
+  private static String decoderInputFile;
+
+  @Before
+  public void setUp() throws SecurityException, FileNotFoundException {
+    decoderOutputFile = "decodeTestOutput.json";
+    decoderInputFile = "testProtoFile";
+    // Attempt to open both the input and output streams for testing.
+    testInputArray =
+        new String[] {
+          "gps_timestamp lat lng alt roll_deg pitch_deg yaw_deg",
+          "1140885738.407125 48.129872 11.582905 10.015034 -177.971817 -3.228588 10.700191",
+          "1140885738.475212 48.129872 11.582905 10.014367 -177.982559 -3.510518 10.743480"
+        };
+    testInput = testInputArray[0] + "\n" + testInputArray[1] + "\n" + testInputArray[2];
+    encoderInput = new ByteArrayInputStream(testInput.getBytes());
+    encoderOutput = new FileOutputStream(decoderInputFile);
+    decoderOutput = new FileOutputStream(decoderOutputFile);
+  }
+
+  @Test
+  /**
+   * Convert the encoded .sstable file (proto format) into json. Basic sanity checks, up to the user
+   * to make sure the output is indeed as expected.
+   */
+  public void validateDecoding() throws IOException, OutOfMemoryError, SecurityException {
+    // The encoder test relies on the encoder to generate the proper proto file to
+    // decode.
+    // First encode data from text to proto format.
+    Encoder.encode(encoderInput, encoderOutput);
+
+    // Now decode the data that was just created in that file.
+    decoderInput = new FileInputStream(decoderInputFile);
+    Decoder.decode(decoderInput, decoderOutput);
+
+    // Open the file and create a String from its contents.
+    File verifyFile = new File(decoderOutputFile);
```

Comment:
Interacting with the file system makes tests slower and flakier, so it's better to avoid it unless the actual behavior you're testing involves the file system. I think you can make these tests work without ever writing out to disk.